include/swift/AST/PrintOptions.h

```diff
@@ -294,6 +275,9 @@ struct PrintOptions {
   /// Whether to print the extensions from conforming protocols.
   bool PrintExtensionFromConformingProtocols = false;
 
+  std::shared_ptr<ShouldPrintChecker> CurrentPrintabilityChecker =
```

Comment:
Stack allocation is usually insufficient for setting the checker because we set in the factory method of `PrintOptions`.